### DISCUSSION TITLE
Zip Export Plugin

### DISCUSSION THREAD

**Discussion by Stage4000** [GrapesJS/grapesjs#3884] [Ideas]
Is there any plans to make the export plugin compatible with the page manager? It appears to only be exporting a single page, is there something im missing or is the plugin just outdated, if so does anyone have a PHP example script that can render the pages on the back end or break down how I can convert the json into an actual page because im pretty lost on that part.

**Comment by Ju99ernaut** [comment]
You can configure the export plugin to export all pages. I don't have an example yet but it's possible.

**Reply by Ju99ernaut** [reply]
`root` can be a function so you can use a configuration like below:
```js
               'grapesjs-plugin-export': {
                    root(ed) { 
                        const all = ed.Pages.getAll();
                        const pages = {};
                        const css = {};
                        all.map(page => {
                            pages[(page.get('name') || page.id) + '.html'] = '<!doctype html>' +
                                '<html lang="en">' +
                                '<head>' +
                                '<meta charset="utf-8">' +
                                '<link rel="stylesheet" href="/css/style-' + page.id + '.css">' +
                                '</head>' +
                                '<body>' +
                                page.getMainComponent().toHTML() +
                                '</body>' +
                                '</html>';
                            css['style-' + page.id + '.css'] = editor.CodeManager.getCode(page.getMainComponent(), 'css')
                        });
                        return {
                            css: {
                                ...css
                            },
                            ...pages
                        }
                    }
                }
```

**Reply by Stage4000** [reply]
Why are you so awesome? Thanks again dude, you really come in for the save twice in a row now. I wish I could move this to Q&A so you got credit for answering my dumb questions.

**Reply by Stage4000** [reply]
I tried to add image exporting to this and im currently getting this error:
Uncaught (in promise) ReferenceError: fetchImagesByStructue is not defined
    at img ((index):167)
    at r.<anonymous> (grapesjs-plugin-export:9)
    at b (grapesjs-plugin-export:9)
    at Generator._invoke (grapesjs-plugin-export:9)
    at Generator.t.<computed> [as next] (grapesjs-plugin-export:9)
    at i (grapesjs-plugin-export:9)
    at grapesjs-plugin-export:9
    
    Is this because I don't have my backend set up for the asset manager yet or is it something else? Its probably something far dumber on my part.

**Reply by Ju99ernaut** [reply]
You'll need to define the fetchImagesByStructure function.
